components/director/internal/domain/runtime/repository_test.go

```diff
@@ -0,0 +1,206 @@
+package runtime_test
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/internal/appdb"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/kyma-incubator/compass/components/director/internal/domain/runtime"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPgRepository_GetByID_ShouldReturnRuntimeModelForRuntimeEntity(t *testing.T) {
+	// given
+	tenantID := uuid.New().String()
+	runtimeID := uuid.New().String()
+
+	time, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	require.NoError(t, err)
+
+	sqlDB, sqlxDB, sqlMock := mockDatabase(t)
+	_ = sqlDB
+
+	rows := sqlmock.NewRows([]string{"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "auth"}).
+		AddRow(runtimeID, tenantID, "Runtime ABC", "Description for runtime ABC", "INITIAL", time, agentAuthStr)
+
+	sqlMock.ExpectQuery(`^SELECT (.+) FROM "public"."runtimes" WHERE "id" = \$1 AND "tenant_id" = \$2$`).
+		WithArgs(runtimeID, tenantID).
+		WillReturnRows(rows)
+
+	ctx := context.WithValue(context.TODO(), appdb.DatabaseCtxKey, sqlxDB)
+
+	pgRepository := runtime.NewPostgresRepository()
+
+	// when
+	modelRuntime, err := pgRepository.GetByID(ctx, tenantID, runtimeID)
+	_ = modelRuntime
+
+	//then
+	assert.NoError(t, err)
+	assert.NoError(t, sqlMock.ExpectationsWereMet())
+}
+
+func TestPgRepository_List_ShouldReturnRuntimeModelsForRuntimeEntities(t *testing.T) {
+	// given
+	tenantID := uuid.New().String()
+
+	time, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	require.NoError(t, err)
+
+	sqlDB, sqlxDB, sqlMock := mockDatabase(t)
+	_ = sqlDB
+
+	rows := sqlmock.NewRows([]string{"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "auth"}).
+		AddRow(uuid.New().String(), tenantID, "Runtime ABC", "Description for runtime ABC", "INITIAL", time, agentAuthStr).
+		AddRow(uuid.New().String(), tenantID, "Runtime XYZ", "Description for runtime XYZ", "INITIAL", time, agentAuthStr)
+
+	sqlMock.ExpectQuery(`^SELECT (.+) FROM "public"."runtimes" WHERE "tenant_id" = \$1$`).
+		WithArgs(tenantID).
+		WillReturnRows(rows)
+
+	ctx := context.WithValue(context.TODO(), appdb.DatabaseCtxKey, sqlxDB)
```

Comment:
to discuss: where we test transactions handling?

Reply:
it will be really nice to have a helper method for decorating context with "db context"